lib/api/expect/cookie.js

```diff
@@ -48,11 +48,18 @@ class ExpectCookie extends BaseExpect {
     promise
       .then((result) => {
         const domainStr = this.cookieDomain ? ` for domain "${this.cookieDomain}"` : '';
-
         if (!result.value) {
           return this.reject(new NotFoundError(`no cookie "${this.cookieName}"${domainStr} was found`));
         }
-        this.resultValue = result ? result.value : {};
+        if (result && result.value) {
+          if (Array.isArray(result.value)) {
+            this.resultValue = result.value[0].value;
```

Comment:
If `result.value` is an empty array, this will throw an error.

Reply:
good catch!

Reply:
I've added a ternary here. Looking at the spec, the likelihood of it being empty for named cookies is low because it should error if it can't return anything. 

I'm not sure how to add a test as it seems to kick the retry code in so any guidance would be appreciated.